ApplicationInsights/JavaScriptSDK/TelemetryItemCreator.ts

```diff
@@ -1,51 +1,74 @@
 import { IPageViewTelemetryInternal } from "../JavaScriptSDK.Interfaces/IPageViewTelemetry";
 
 export interface ITelemetryItemCreator {
-    create(pageView: IPageViewTelemetryInternal, baseType: string, envelopeName: string, customProperties?: { [key: string]: any }): ITelemetryItem
+    create(item: IPageViewTelemetryInternal | PageViewPerformance, baseType: string, envelopeName: string, customProperties?: { [key: string]: any }): ITelemetryItem
 }
 
 export class TelemetryItemCreator implements ITelemetryItemCreator {
     private static creator = new TelemetryItemCreator();
 
-    public static createItem(pageView: IPageViewTelemetryInternal, baseType: string, envelopeName: string, customProperties?: { [key: string]: any }, systemProperties?: { [key: string]: any }): ITelemetryItem {
-        if (CoreUtils.isNullOrUndefined(pageView) ||
+    /**
+     * A wrapper for telemetry item create. Ensures valid parameters
+     * @param item domain specific properties; part B
+     * @param baseType telemetry item type. ie PageViewData
+     * @param envelopeName name of the envelope. ie Microsoft.ApplicationInsights.<instrumentation key>.PageView
+     * @param customProperties user defined custom properties; part C
+     * @param systemProperties system properties that are added to the context; part A
+     * @returns ITelemetryItem that is sent to channel
+     */
+    public static createItem(item: IPageViewTelemetryInternal | PageViewPerformance,
+        baseType: string,
+        envelopeName: string,
+        customProperties?: { [key: string]: any },
+        systemProperties?: { [key: string]: any }): ITelemetryItem {
+        if (CoreUtils.isNullOrUndefined(item) ||
             CoreUtils.isNullOrUndefined(baseType) ||
             CoreUtils.isNullOrUndefined(envelopeName)) {
             throw Error("pageView doesn't contain all required fields");
         };
 
-        return TelemetryItemCreator.creator.create(pageView, baseType, envelopeName, customProperties, systemProperties);
+        return TelemetryItemCreator.creator.create(item, baseType, envelopeName, customProperties, systemProperties);
     }
 
-    create(pageView: IPageViewTelemetryInternal, baseType: string, envelopeName: string, customProperties?: { [key: string]: any }, systemProperties?: { [key: string]: any }): ITelemetryItem {
+    /**
+     * Create a telemetry item that the 1DS channel understands
+     * @param item domain specific properties; part B
+     * @param baseType telemetry item type. ie PageViewData
+     * @param envelopeName name of the envelope. ie Microsoft.ApplicationInsights.<instrumentation key>.PageView
+     * @param customProperties user defined custom properties; part C
+     * @param systemProperties system properties that are added to the context; part A
+     * @returns ITelemetryItem that is sent to channel
+     */
+    create(item: IPageViewTelemetryInternal | PageViewPerformance,
```

Comment:
can we track two stories, one for serialization and other for inheriting off generated type from applicationinsights-home